File: pharmcrew-admin-ng/src/app/domain/me/me.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from 'src/app/http.service';
import { MeResp } from './me-resp';
import { PcResp } from 'src/app/pc-resp';
import { UpdateMeReq } from './update-me-req';
import { FindLoginIdReq } from 'src/app/domain/me/find-login-id-req';
import { FindLoginIdResp } from 'src/app/domain/me/find-login-id-resp';
import { ResetPasswordReq } from './reset-password-req';

@Injectable({
  providedIn: 'root'
})
export class MeService extends HttpService {
  findLoginId(req: FindLoginIdReq) {
    return this.get<FindLoginIdResp>('/loginId', req);
  }
  resetPassword(req: ResetPasswordReq) {
    return this.put<PcResp>('/password/reset', req);
  }
  me() {
    return this.get<MeResp>('/me');
  }
  updateMe(req: UpdateMeReq) {
    return this.put<PcResp>('/me', req);
  }
}

File: pharmcrew-admin/src/main/java/kr/ant/kpa/pharmcrew/service/MeService.java
package kr.ant.kpa.pharmcrew.service;

import kr.ant.kpa.pharmcrew.resp.PcResp;
import kr.ant.kpa.pharmcrew.resp.me.FindLoginIdResp;
import kr.ant.kpa.pharmcrew.resp.me.MeResp;

public interface MeService {
    FindLoginIdResp findLoginId(String name, String email);
    PcResp resetPassword(String name, String loginId, String email);
    MeResp me(long myId);
    PcResp updateMe(long myId, String password, String name, String email, String tel, String phone);
}
